urlcdab7d8364_jrwalsh_CycTools_tgz-pJ8-edu_iastate_cyctools_MainJ8/src/edu/iastate/cyctools/externalSourceCode/DiffMatchPatch.java

```diff
@@ -0,0 +1,2535 @@
+                index1 += aDiff.text.length();
+              }
+            }
+          }
+        }
+      }
+      x++;
+    }
+    // Strip the padding off.
+    text = text.substring(nullPadding.length(), text.length()
+        - nullPadding.length());
+    return new Object[]{text, results};
+  }
+
+  /**
+   * Add some padding on text start and end so that edges can match something.
+   * Intended to be called only from within patch_apply.
+   * @param patches Array of Patch objects.
+   * @return The padding string added to each side.
+   */
+  public String patch_addPadding(LinkedList<Patch> patches) {
+    short paddingLength = this.Patch_Margin;
+    String nullPadding = "";
+    for (short x = 1; x <= paddingLength; x++) {
+      nullPadding += String.valueOf((char) x);
+    }
+
+    // Bump all the patches forward.
+    for (Patch aPatch : patches) {
+      aPatch.start1 += paddingLength;
+      aPatch.start2 += paddingLength;
+    }
+
+    // Add some padding on start of first diff.
+    Patch patch = patches.getFirst();
+    LinkedList<Diff> diffs = patch.diffs;
+    if (diffs.isEmpty() || diffs.getFirst().operation != Operation.EQUAL) {
+      // Add nullPadding equality.
+      diffs.addFirst(new Diff(Operation.EQUAL, nullPadding));
+      patch.start1 -= paddingLength;  // Should be 0.
+      patch.start2 -= paddingLength;  // Should be 0.
+      patch.length1 += paddingLength;
+      patch.length2 += paddingLength;
+    } else if (paddingLength > diffs.getFirst().text.length()) {
+      // Grow first equality.
+      Diff firstDiff = diffs.getFirst();
+      int extraLength = paddingLength - firstDiff.text.length();
+      firstDiff.text = nullPadding.substring(firstDiff.text.length())
+          + firstDiff.text;
+      patch.start1 -= extraLength;
+      patch.start2 -= extraLength;
+      patch.length1 += extraLength;
+      patch.length2 += extraLength;
+    }
+
+    // Add some padding on end of last diff.
+    patch = patches.getLast();
+    diffs = patch.diffs;
+    if (diffs.isEmpty() || diffs.getLast().operation != Operation.EQUAL) {
+      // Add nullPadding equality.
+      diffs.addLast(new Diff(Operation.EQUAL, nullPadding));
+      patch.length1 += paddingLength;
+      patch.length2 += paddingLength;
+    } else if (paddingLength > diffs.getLast().text.length()) {
+      // Grow last equality.
+      Diff lastDiff = diffs.getLast();
+      int extraLength = paddingLength - lastDiff.text.length();
+      lastDiff.text += nullPadding.substring(0, extraLength);
+      patch.length1 += extraLength;
+      patch.length2 += extraLength;
+    }
+
+    return nullPadding;
+  }
+
+  /**
+   * Look through the patches and break up any which are longer than the
+   * maximum limit of the match algorithm.
+   * Intended to be called only from within patch_apply.
+   * @param patches LinkedList of Patch objects.
+   */
+  public void patch_splitMax(LinkedList<Patch> patches) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 22. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.